daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/DPath.scala

```diff
@@ -160,6 +161,12 @@ final class RuntimeExpressionDPath[T <: AnyRef](qn: NamedQName, tt: NodeInfo.Kin
             case _ => doSDE(e, ps)
           }
         }
+        case ParserNonBlocking => {
+          e match {
+            case nd: InfosetNoDataException => throw new InfosetSelfReferencingException(state.dState.currentNode)
```

Comment:
Rather than throwing, should this just creat the exception and the call doSDE? This way doSDE can convert it to an SDE so the error can be handled appropriately.

Reply:
Will do.